gbasis/evals/density.py

```diff
@@ -243,21 +255,34 @@ def evaluate_deriv_density(
                 num_occurence = comb(total_l_x, l_x) * comb(total_l_y, l_y) * comb(total_l_z, l_z)
                 orders_one = np.array([l_x, l_y, l_z])
                 orders_two = orders - orders_one
-                density = evaluate_deriv_reduced_density_matrix(
-                    orders_one,
-                    orders_two,
-                    one_density_matrix,
-                    basis,
-                    points,
-                    transform=transform,
-                    coord_type=coord_type,
-                )
+                if any(orders_one > 2) or any(orders_two > 2):
+                    density = evaluate_deriv_reduced_density_matrix(
+                            orders_one,
+                            orders_two,
+                            one_density_matrix,
+                            basis,
+                            points,
+                            transform=transform,
+                            coord_type=coord_type,
+                            deriv_type='general',
+                        )
+                else:
+                    density = evaluate_deriv_reduced_density_matrix(
```

Comment:
If there are no benefits to using the general method for the orders less than or equal to 2, it may be better to make it the default behaviour.

Reply:
Actually, I change my mind because the `deriv_type` seems to be used quite frequently throughout the code.